woocommerce/woocommerce-ios
# Refactor store stats data / empty state UI and always show the stats views

Part of #5745 


### Description


Based on the design decision in this thread p1642555588011500/1639990681.280100-slack-CGPNUU63E, we now always show the visitor/order/conversion stats. In addition, when visitor stats are unavailable due to Jetpack or API reasons, we are introducing the third state of stats view - redacted (without the Jetpack logo). The redacted states apply to not only visitor stats, but also conversion stats. Before making the final design changes, I decided to refactor the stats views with a new view class `StoreStatsDataOrRedactedView` to encapsulate the "data or redacted" states. `StoreStatsDataOrRedactedView` now has three states: `data`, `redacted`, and `redactedDueToJetpack`.

The original `StoreStatsSiteVisitEmptyView` for the visitor stats empty state was renamed to `StoreStatsEmptyView` with two changes:
- `showJetpackImage: Bool` is set to show the Jetpack logo or not
- Auto Layout constraints are updated when the `myStoreTabUpdates` feature flag is enabled

Finally, `StoreStatsV4PeriodViewController` was updated to use `StoreStatsDataOrRedactedView` for the order/visitor/conversion stats. The updates on visitor and conversion stats states will be in a future PR due to the PR size.

### Testing instructions


The only visual change should be showing visitor and conversion stats all the time.

Prerequisite: the site has at least one order today
- Launch the app --> all stats should be shown
- Select an hour in the Today tab --> all stats should be shown

- [x] @jaclync tests JCP sites: all stats should be shown where the visitor stats still show the redacted state with a Jetpack logo

### Screenshots


JCP site | today tab - selected | today tab
-- | -- | --
![Simulator Screen Shot - iPhone 11 - 2022-01-20 at 16 00 54](https://user-images.githubusercontent.com/1945542/150297378-f5c71568-bb95-413c-8d54-8082256e5e6b.png) | ![Simulator Screen Shot - iPhone 11 - 2022-01-20 at 16 01 13](https://user-images.githubusercontent.com/1945542/150297384-35baede8-5ecf-4f9f-925b-ed6ad5550748.png) | ![Simulator Screen Shot - iPhone 11 - 2022-01-20 at 16 01 18](https://user-images.githubusercontent.com/1945542/150297391-5efe0fdd-9ed8-44a7-8666-4d136ee80ff1.png)


---
